Marlin/Marlin_main.cpp

```diff
@@ -5067,46 +5067,65 @@ inline void gcode_G28() {
-     *     T  = do not calibrate tower angles with 7+ point calibration
+     * G33 - Delta '1-4-7-point' auto calibration
+     *       Calibrate height, endstops, delta radius, and tower angles.
+     *
+     * Parameters:
+     *
+     *   P  Number of probe points:
+     *
+     *      P1     Probe center and set height only.
+     *      P2     Probe center and towers. Set height, endstops, and delta radius.
+     *      P3     Probe all positions: center, towers and opposite towers. Set all.
+     *      P4-P7  Probe all positions at different locations and average them.
+     *
+     *   A  Abort delta height calibration after 1 probe (P1)
+     *
+     *   O  Use opposite tower points instead of tower points (P2 only)
+     *
+     *   T  Don't calibrate tower angle corrections (for P3-P7)
+     *
+     *   V  Verbose level:
+     *      V0  Dry-run mode. Report settings and probe results. No calibration.
+     *      V1  Report settings
+     *      V2  Report settings and probe results
      */
     inline void gcode_G33() {
 
-      stepper.synchronize();
+      if (axis_unhomed_error(true, true, true)) return;
 
-      #if PLANNER_LEVELING
-        set_bed_leveling_enabled(false);
-      #endif
+      const int8_t p_value = code_seen('P') ? code_value_int() : DELTA_CALIBRATION_DEFAULT_POINTS;
+      if (!WITHIN(p_value, -7, 7)) {
+        SERIAL_PROTOCOLLNPGM("?(P)oints is implausible (-7 to 7).");
+        return;
+      }
 
-      int8_t pp = (code_seen('P') ? code_value_int() : DELTA_CALIBRATION_DEFAULT_POINTS),
-             probe_mode = (WITHIN(pp, 1, 7) ? pp : DELTA_CALIBRATION_DEFAULT_POINTS);
+      const int8_t verbose_level = code_seen('V') ? code_value_byte() : 1;
+      if (!WITHIN(verbose_level, 0, 2)) {
+        SERIAL_PROTOCOLLNPGM("?(V)erbose Level is implausible (0-2).");
+        return;
+      }
 
-      probe_mode = (code_seen('A') && probe_mode == 1 ? -probe_mode : probe_mode);
-      probe_mode = (code_seen('O') && probe_mode == 2 ? -probe_mode : probe_mode);
-      probe_mode = (code_seen('T') && probe_mode > 2 ? -probe_mode : probe_mode);
-      
-      int8_t verbose_level = (code_seen('V') ? code_value_byte() : 1);
+      int8_t probe_mode = WITHIN(p_value, 1, 7) ? p_value : DELTA_CALIBRATION_DEFAULT_POINTS;
+      const char negating_parameter = probe_mode == 1 ? 'A' : probe_mode == 2 ? 'O' : 'T';
+      if (code_seen(negating_parameter) && code_value_bool()) probe_mode = -probe_mode;
+
+      stepper.synchronize();
 
-      if (!WITHIN(verbose_level, 0, 2)) verbose_level = 1;
+      #if PLANNER_LEVELING || ENABLED(AUTO_BED_LEVELING_UBL)
+        set_bed_leveling_enabled(false);
+      #endif
 
-      gcode_G28();
+      const uint8_t probe_points = abs(probe_mode);
+      const bool pp_height_only = probe_points == 1,
+                 pp_center_and_towers = probe_points == 2,
+                 pp_all_positions = probe_points == 3,
+                 pp_5_points = probe_points == 5,
```

Comment:
suggestion if you prefer descriptive names:
==1 pp_center_only
==2 pp_towers_or_opposites
==3 pp_towers_and_opposites
(==4 pp_all_single_circle)
==5 pp_all_double_circle
==6 pp_all_triple_circle
==7 pp_all_quadruple_circle
> 2 pp_with_tower_angles
> 3 pp_with_intermediates
> 4 pp_with_multi_circles